docs/cookbook/recipe_sonata_admin_without_user_bundle.rst

```diff
@@ -0,0 +1,452 @@
+SonataAdmin without FOSUserBundle/SonataUserBundle
+==================================================
+
+Here we will explain how can you use SonataAdmin with Symfony Guard instead
+of using FOSUserBundle or SonataUserBundle.
+
+Code used in this short guide can be found `here <https://github.com/kunicmarko20/guard-talk-example>`__
+with support for different Symfony versions.
+
+.. note::
+    This article assumes you are using Symfony 4.
+
+The recipe
+----------
+
+Things we will need to do:
+
+- User Entity
+- UserProvider
+- LoginForm
+- LoginAuthenticator
+- LoginController
+- Setup firewall in `security.yaml`
+- Login template resembling Sonata style
+
+User Entity
+^^^^^^^^^^^
+
+This represents your security user, you can read more about it `here <http://symfony.com/doc/current/security/entity_provider.html#create-your-user-entity>`__.
+
+.. code-block:: php
+
+    namespace App\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Security\Core\User\UserInterface;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="user")
+     */
+    class User implements UserInterface
+    {
+        /**
+         * @ORM\Id
+         * @ORM\Column(type="integer")
+         * @ORM\GeneratedValue(strategy="AUTO")
+         */
+        private $id;
+
+        /**
+         * @ORM\Column(name="email", type="string", unique=true)
+         */
+        private $email;
+
+        /**
+         * @ORM\Column(name="password", type="string", nullable=true)
+         */
+        private $password;
+
+        /**
+         * @ORM\Column(name="roles", type="json_array")
+         */
+        private $roles;
+
+        //...
+    }
+
+UserProvider
+^^^^^^^^^^^^
+
+This represents your user provider, it will be used to load your security users, read
+more about it `here <http://symfony.com/doc/current/security/custom_provider.html#create-a-user-provider>`__.
+
+.. code-block:: php
+
+    namespace App\Security;
+
+    use App\Entity\User;
+    use Doctrine\ORM\EntityManagerInterface;
+    use Symfony\Component\Security\Core\User\UserInterface;
+    use Symfony\Component\Security\Core\User\UserProviderInterface;
+    use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+
+    class UserProvider implements UserProviderInterface
+    {
+        /**
+         * @var EntityManagerInterface
+         */
+        private $entityManager;
+
+        public function __construct(EntityManagerInterface $entityManager)
+        {
+            $this->entityManager = $entityManager;
+        }
+
+        public function loadUserByUsername($email)
+        {
+            $user = $this->findUserBy(['email' => $email]);
+
+            if (!$user) {
+                throw new UsernameNotFoundException(
+                    sprintf(
+                        'User with "%s" email does not exist.',
+                        $email
+                    )
+                );
```

Comment:
```php
<?php
throw new UsernameNotFoundException(sprintf(
    'User with "%s" email does not exist.',
    $email
));
```

Splitting only one function will decrease the line length

Reply:
fixed.